### PR TITLE
Travis ci integration

### PR DESCRIPTION
**Pull request by realityforge**
This adds basic TravisCI integration. Notifications on build failures/recovery have been disabled but other options are possible as discussed in google/elemental2#82

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# NOTE: we don't include OS- or IDE/editor-specific files (e.g. Windows' Thumbs.db,
+# OSX's .DS_Store, Vim's *.swp, Emacs' *~, Visual Studio Code's .vscode, etc.) on-purpose.
+# Those patterns should go in a global gitignore or repository-specific excludes.
+# See https://help.github.com/articles/ignoring-files/ for how to configure your environment.
+# See https://github.com/github/gitignore/tree/master/Global for a list of global ignore rules.
+/bazel-*
```
Comment (jDramaix) [inline .gitignore]: This should be part of another PR as it's not related to TravisCI

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  - echo "startup --host_jvm_args=-Xmx2500m" >> .bazelrc
+  - echo "startup --host_jvm_args=-Xms2500m" >> .bazelrc
+  # Avoid progress indicators that can blow out log which makes using WEB UI difficult
+  - echo "common --noshow_progress" >> .bazelrc
+  - echo "common --noshow_loading_progress" >> .bazelrc
+  # This is so we understand failures better
+  - echo "build --verbose_failures" >> .bazelrc
+
+script:
+  - bazel build //java/jsinterop/generator/closure:ClosureJsinteropGenerator_deploy.jar
```
Comment (jDramaix) [inline .travis.yml]: bazel build ...

File: .gitignore (hunk 1 above)
Comment (realityforge) [reply .gitignore]: Fixed - moved to https://github.com/google/jsinterop-generator/pull/14

File: .travis.yml (hunk 2 above)
Comment (realityforge) [reply .travis.yml]: Fixed

**Comment by realityforge** [comment]
Force pushed another update to move to using `./bazel_build_test.sh` and bazel `0.23.0`

**Comment by realityforge** [comment]
Update to match the style of other repos

**Comment by realityforge** [comment]
Tah
